src/Sticky.jsx

```diff
@@ -289,6 +289,13 @@ class Sticky extends Component {
         if (prevState.status !== this.state.status && this.props.onStateChange) {
             this.props.onStateChange({status: this.state.status});
         }
+
+        // check if we are up-to-date, is triggered in case of scroll restoration
+        if (this.props.top !== prevState.top) {
```

Comment:
any way to write a test for this?

Reply:
I don’t feel capable to add a test as I’m unable to grasp the general idea of how these are set up. I also believe it’s more an edge case that can be argued about logically. Do you ever expect `this.props.top` to be unequal to `prevState.top` in the `componentDidUpdate` event? I guess, no. But if this ever happens you might agree, that an “update” is necessary.

Reply:
@roderickhsiao @hankhsiao care to chime in here?

Reply:
Code looks fine to me.

Reply:
You have missed the case when `this.props.top` is the name of class. It causes infinite updates by calling `this.setState` again and again.

Reply:
Hi @unfalse, do you mind elaborating a bit on what you mean by “is the name of class”? Please provide a small example, if possible.

Reply:
Hi @m99coder
The `Sticky` component allows us to provide a `top` property of type string (class of an element). If the top property is a string, it performs a `querySelector` and calculates the top position [here](https://github.com/yahoo/react-stickynode/blob/66d8ec8199adfca1d1eff40eadc7e631267b7655/src/Sticky.jsx#L71). Following this, it sets a number to the state. As a result, the `top` property in the state will always be a number, while `this.props.top` will continue to be a string.

If we provide the `top` property as a string, then the following comparison will always be true as it is comparing a string with a number: `if (this.props.top !== prevState.top) {`. This will result in an infinite loop: `this.updateInitialDimension() -> this.setState() -> componentDidUpdate() -> this.updateInitialDimension() ....`

I think it make sense to compare the current state with the prev state `if (this.state.top !== prevState.top) {`

Reply:
Created a new PR with exactly your suggestion here: https://github.com/yahoo/react-stickynode/pull/939